sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -103,7 +103,7 @@ object DataType {
 
   /** Given the string representation of a type, return its DataType */
   private def nameToType(name: String): DataType = {
-    val FIXED_DECIMAL = """decimal\(\s*(\d+)\s*,\s*(\d+)\s*\)""".r
+    val FIXED_DECIMAL = """decimal\(\s*(\d+)\s*,\s*(\-?\d+)\s*\)""".r
```

Comment:
Why change this? `scale` must not be negative because `0 <= scale <= precision`.